File: plot_coincident_pulses.m
%% plot 5 pulses in a coincidence

% Define the index of the coincidence that you want to plot
coincidence_index = 1;

% Load the data
filename = 'test/coincidences.h5';
info = h5info(filename);
dataset_name = '/voltage_pulses';
coincident_pulses = h5read(filename, dataset_name);
% matlab uses column-major layout by default
coincident_pulses = permute(coincident_pulses, [3 2 1]);

% Plot the data
figure;
hold on;
for i = 1:5
    plot(squeeze(coincident_pulses(coincidence_index, i, :)), 'DisplayName', sprintf('Channel %d', i-1));
end
xlabel('Time (ns) / 2');
ylabel('Voltage (V)');
legend;
hold off;

File: plot_coincident_pulses.py
'''
Description: 
Author: Ming Fang
Date: 2024-05-20 14:48:33
LastEditors: Ming Fang
LastEditTime: 2024-05-20 14:50:17
'''
import h5py
from matplotlib import pyplot as plt

if __name__ == "__main__":
    coincidence_index = 1
    # Load the data
    with h5py.File("test/coincidences.h5", "r") as f:
        coincident_pulses = f["voltage_pulses"][:]

    # Plot the data
    fig, ax = plt.subplots()
    for i in range(5):
        ax.plot(coincident_pulses[coincidence_index, i, :], label=f"Channel {i}")
    ax.set_xlabel("Time (ns) / 2")
    ax.set_ylabel("Voltage (V)")
    ax.legend()
    plt.show()
